### PR TITLE
Refine go planner

### PR DESCRIPTION
**Pull request by jievince**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Issue(s) number: 
Close https://github.com/vesoft-inc/nebula/issues/4709
## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by Shylock-Hg — APPROVED** [review]
LGTM, but better do it in Optimizer.

**Comment by jievince** [comment]
> LGTM, but better do it in Optimizer.
It's a little hard to do it in optimizer.

File: src/graph/planner/ngql/GoPlanner.cpp [hunk 1]
```diff
@@ -454,43 +474,40 @@ SubPlan GoPlanner::nStepsPlan(SubPlan& startVidPlan) {
...
+  PlanNode* loopDep = startVidPlan.root;
   if (!goCtx_->joinInput && goCtx_->limits.empty()) {
     auto* gd = GetDstBySrc::make(qctx, start, goCtx_->space.id);
     gd->setSrc(goCtx_->from.src);
     gd->setEdgeTypes(buildEdgeTypes());
     gd->setInputVar(goCtx_->vidsVar);
-    gd->setColNames({kDst});
-    scan = gd;
+    gd->setColNames({goCtx_->dstIdColName});
+    auto* dedup = Dedup::make(qctx, gd);
```
Comment (nevermore3) [inline src/graph/planner/ngql/GoPlanner.cpp:487]: The result of getDstBySrc operator has been deduplicated, why need dedup operator

Comment (jievince) [reply src/graph/planner/ngql/GoPlanner.cpp:487]: The operator is only de-duplicated on the storaged side. When graphd merges the results from multiple storaged, it needs to be de-duplicated again.

Comment (nevermore3) [reply src/graph/planner/ngql/GoPlanner.cpp:487]: got it

**Review by nevermore3 — APPROVED** [review]

**Comment by shanlai** [comment]
3.3 rc checked

**Pull request merged**